app/code/core/Mage/Customer/Model/Config/Forms.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * OpenMage
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@magento.com so we can send you a copy immediately.
+ *
+ * @category    Mage
+ * @package     Mage_Customer
+ * @copyright  Copyright (c) 2006-2020 Magento, Inc. (http://www.magento.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Used in creating options for use_in_forms selection
+ *
+ */
+class Mage_Customer_Model_Config_Forms
+{
+
+    /**
+     * Options getter
+     *
+     * @return array
+     */
+    public function toOptionArray()
```

Comment:
I have custom form types, is it possible to define these in `app\code\core\Mage\Customer\etc\config.xml`?

Reply:
Would it maybe be better to set up an event to hook into? I'm thinking about the best way a module could define their own form types.

Reply:
Using event is ok, but I like the config method. An example is product types:
https://github.com/OpenMage/magento-lts/blob/35acb46f0385484ec45397cc53b379ecf1434ea6/app/code/core/Mage/Catalog/etc/config.xml#L479-L487

and product custom option:
https://github.com/OpenMage/magento-lts/blob/35acb46f0385484ec45397cc53b379ecf1434ea6/app/code/core/Mage/Catalog/etc/config.xml#L531-L537

which I have used to add more options in the HTML select element.

Reply:
@kiatng thanks for the examples, I was thinking about it more and agree the config.xml method is indeed better.

Reply:
Ref https://github.com/OpenMage/magento-lts/blob/35acb46f0385484ec45397cc53b379ecf1434ea6/app/code/core/Mage/Adminhtml/Model/System/Config/Source/Product/Options/Type.php#L34

I made a sample code:
```php
class Mage_Adminhtml_Model_System_Config_Source_Customer_Form
{
    const CUSTOMER_FORM_PATH = 'global/customer/form';

    public function toOptionArray()
    {
        $helper = Mage::helper('customer');
        foreach (Mage::getConfig()->getNode(self::CUSTOMER_FORM_PATH)->children() as $form) {
            $labelPath = self::CUSTOMER_FORM_PATH . '/' . $form->getName() . '/label';
            $forms[] = [
                'label' => $helper->__((string) Mage::getConfig()->getNode($labelPath)),
                'value' => $form->getName()
            ];
        }

        return $forms;
    }
}
```

The `config.xml` looks like this:
```xml
    <global>
        <customer>
            <form>
                <adminhtml_checkout translate="label" module="customer">
                    <label>Adminhtml Checkout</label>
                </adminhtml_checkout>
                <adminhtml_customer translate="label" module="customer">
                    <label>Adminhtml Customer</label>
                </adminhtml_customer>
                <!-- more forms -->
            </form>
        </customer>
    </global>
```

Reply:
That code sample doesn't consider the module attribute. Corrected version:
```php
class Mage_Adminhtml_Model_System_Config_Source_Customer_Form
{
    const CUSTOMER_FORM_PATH = 'global/customer/form';

    public function toOptionArray()
    {
        foreach (Mage::getConfig()->getNode(self::CUSTOMER_FORM_PATH)->children() as $form) {
            $labelPath = self::CUSTOMER_FORM_PATH . '/' . $form->getName() . '/label';
            $moduleName = $form->getAttribute('module') ?? 'customer';
            $forms[] = [
                'label' => Mage::helper($moduleName)->__((string) Mage::getConfig()->getNode($labelPath)),
                'value' => $form->getName()
            ];
        }

        return $forms;
    }
}
```
Note that `$form->getAttribute('module')` can return `null`.

Reply:
@kiatng thanks for the implementation! 

Do you think this XML is better? Since we need to also define `customer_address` forms it keeps all forms under one node, `<form>`.

```xml
<global>
   <form>
      <customer>
         <adminhtml_checkout translate="label" module="customer">
            <label>Adminhtml Checkout</label>
         </adminhtml_checkout>
         <adminhtml_customer translate="label" module="customer">
            <label>Adminhtml Customer</label>
         </adminhtml_customer>
         <!-- more forms -->
      </customer>
      <customer_address>
         <adminhtml_customer_address translate="label" module="customer">
            <label>Adminhtml Customer Address</label>
         </adminhtml_customer_address>
         <customer_address_edit translate="label" module="customer">
            <label>Customer Address Edit</label>
         </customer_address_edit>
         <!-- more forms -->
      </customer_address>
   </form>
</global>
```

I could also move the source models to:
```
Mage_Adminhtml_Model_System_Config_Source_Form_Customer
Mage_Adminhtml_Model_System_Config_Source_Form_Customer_Address
```

to match.

Reply:
Ref https://github.com/OpenMage/magento-lts/blob/18e68a223dffc3d8dec6bb3e7ac485a6338baacd/app/code/core/Mage/Customer/Model/Form.php#L35-L35

and https://github.com/OpenMage/magento-lts/blob/18e68a223dffc3d8dec6bb3e7ac485a6338baacd/app/code/core/Mage/Catalog/etc/config.xml#L620-L621

Use the module name like `<catalog>`:

```xml
<global>
   <eav>
     <form>
        <customer>
           <!-- more forms -->
        </customer>
        <customer_address>
           <!-- more forms -->
        </customer_address>
     </form>
   </eav>
</global>
```
or continue with the format `<eav_attributes>`:
```xml
<global>
   <eav_forms>
      <customer>
         <!-- more forms -->
      </customer>
      <customer_address>
         <!-- more forms -->
      </customer_address>
   </eav_forms>
</global>
```
Which do you prefer? I'd go with the latter.

Reply:
I like the latter as well :+1: